src/metadata_validator.py

```diff
@@ -345,17 +346,56 @@ def validate_relationship(self, data_record):
+                    parents = [item for item in data_record_parent_nodes if item[PARENT_TYPE] == parent_type and item[PARENT_ID_NAME] == parent_id_property ]
+                    if len(parents) > 1:
+                        error_type = "One-to-one relationship conflict" if rel_type == "one_to_one" else "Many-to-one relationship conflict"
+                        parent_node_ids = [item[PARENT_ID_VAL] for item in parents]
+                        result[ERRORS].append(create_error(error_type, 
+                                    f'"{msg_prefix}": associated with multiple “{parent_type}” nodes: {json.dumps(parent_node_ids)}.'))
+                    checked_multi_parents = True
+
+            if (parent_type, parent_id_property, parent_id_value) not in parent_nodes:
                 released_parent = self.mongo_dao.search_released_node(data_common, parent_type, parent_id_value)
                 if not released_parent:
                     result[ERRORS].append(create_error("Related node not found", f'Related node “{parent_type}” [“{parent_id_property}”: “{parent_id_value}"] not found.'))
+                else:
+                    if rel_type == "one_to_one":
+                        # check released and current children by released parent
+                        child_node_ids = self.get_unique_child_node_ids(data_common, node_type, parent_node, submission_id)
+                        if child_node_ids and len(child_node_ids) > 1:
+                            result[ERRORS].append(create_error("One-to-one relationship conflict", 
+                                    f'"{msg_prefix}": associated node “{parent_type}”: “{parent_id_value}" has multiple nodes associated: {json.dumps(child_node_ids)}.'))
+            else: 
+                if rel_type == "one_to_one":
+                    # check released and current children by current parent
+                    child_node_ids = self.get_unique_child_node_ids(data_common, node_type, parent_node, submission_id)
+                    if child_node_ids and len(child_node_ids) > 1:
+                        result[ERRORS].append(create_error("One-to-one relationship conflict", 
+                                    f'"{msg_prefix}": associated node “{parent_type}”: “{parent_id_value}" has multiple nodes associated: {json.dumps(child_node_ids)}.'))
 
         if len(result[WARNINGS]) > 0:
             result["result"] = STATUS_WARNING
 
         if len(result[ERRORS]) == 0 and len(result[WARNINGS]) == 0:
             result["result"] = STATUS_PASSED
         return result
+    
+    def get_unique_child_node_ids(self, data_common, node_type, parent_node, submission_id):
+        children = self.mongo_dao.get_nodes_by_parent_prop(node_type, parent_node, submission_id)
+        if not children:
+            return None
+        child_id_list = list(set([item[NODE_ID] for item in children]))
+        if len(child_id_list) > 1:
+            return child_id_list
+        else:
```

Comment:
What if parent node exist both in current submission and release collection? I think we need to collect both child lists and combine them (remove duplicated if exists)

Reply:
If there more than one unique children in current submission,  don't need to check any release children.   otherwise, combine with release children and get final uniques children ids.